src/change_stream.ts

```diff
@@ -532,22 +531,16 @@ function createChangeStreamCursor(
 
   relayEvents(changeStreamCursor, self, ['resumeTokenChanged', 'end', 'close']);
 
-  if (self.listenerCount(ChangeStream.CHANGE) > 0) {
-    changeStreamCursor.on(ChangeStreamCursor.DATA, function (change) {
-      processNewChange(self, change);
-    });
-  }
-
-  changeStreamCursor.on(ChangeStream.ERROR, function (error) {
-    processError(self, error);
-  });
+  const stream = changeStreamCursor.stream(self.streamOptions);
+  changeStreamCursor[kCursorStream] = stream;
+  relayEvents(stream, self, ['end', 'close']);
 
-  if (self.pipeDestinations) {
-    const cursorStream = changeStreamCursor.stream(self.streamOptions);
-    for (const pipeDestination of self.pipeDestinations) {
-      cursorStream.pipe(pipeDestination);
-    }
+  if (self.listenerCount(ChangeStream.CHANGE) > 0) {
+    stream.on(CursorStream.DATA, change => processNewChange(self, change));
   }
+  stream.on(ChangeStream.ERROR, error => processError(self, error));
+  // for internal use only, allows for triggering of mock errors on the internal stream
+  self.on('streamError', event => stream.emit('error', event));
```

Comment:
Is it possible to make this change without introducing a new custom error event? It seems like we should be able to do something like stub a method on `CursorStream` with a fake which emits the error?

Reply:
The problem is we need some way of accessing the `processError` function (which is not exported from `change_stream.ts`). I added an `@internal` method called `triggerError` to do this for now, but open to advice about better ways of achieving this.

Reply:
Oops missed this comment, I left my suggestion below in the test